spec/models/concerns/user/low_balance_fraud_check_spec.rb

```diff
@@ -108,4 +108,90 @@
+            @creator.update!(user_risk_state: "compliant")
+            @creator.update!(user_risk_state: "flagged_for_fraud")
+            @creator.update!(user_risk_state: "on_probation")
+
+            allow(@creator).to receive(:unpaid_balance_cents).and_return(100_00)
+            expect { @creator.restore_risk_state_if_balance_recovered! }
+              .to change { @creator.reload.user_risk_state }.from("on_probation").to("compliant")
+
+            compliant_comment = @creator.comments.where(comment_type: Comment::COMMENT_TYPE_COMPLIANT).order(created_at: :desc).first
+
+            expect(compliant_comment.author_name).to eq("LowBalanceFraudCheck")
+            expect(compliant_comment.content).to include("Marked compliant automatically", "balance has recovered to $100")
+            expect(@creator.reload.refunds_disabled?).to eq(false)
+          end
+        end
+      end
+
+      context "when user was never compliant" do
+        with_versioning do
+          it "reverts to not_reviewed and creates a comment when balance is at or above $100" do
+            @creator.update!(user_risk_state: "not_reviewed")
+            @creator.update!(user_risk_state: "on_probation")
+
+            allow(@creator).to receive(:unpaid_balance_cents).and_return(100_00)
+            expect { @creator.restore_risk_state_if_balance_recovered! }
+            .to change { @creator.reload.user_risk_state }.from("on_probation").to("not_reviewed")
+
+            comment = @creator.comments.where(comment_type: Comment::COMMENT_TYPE_NOTE).order(created_at: :desc).first
+            expect(comment.author_name).to eq("LowBalanceFraudCheck")
+            expect(comment.content).to include("Risk state reverted to \"Not Reviewed\" automatically", "balance has recovered to $100")
+          end
+        end
+      end
+    end
+
+    context "when user is on probation but not for low balance" do
+      it "does not mark user as compliant if balance is above $100" do
+        @creator.comments.create!(comment_type: Comment::COMMENT_TYPE_ON_PROBATION, author_name: "LowBalanceFraudCheck", content: "Probated (payouts suspended) automatically on #{Time.current.to_fs(:formatted_date_full_month)} because of suspicious refund activity", created_at: 1.month.ago)
+        @creator.comments.create!(comment_type: Comment::COMMENT_TYPE_COMPLIANT, author_name: "LowBalanceFraudCheck", content: "Marked compliant automatically on #{Time.current.to_fs(:formatted_date_full_month)} as balance has recovered to $100", created_at: 1.day.ago)
+        @creator.put_on_probation!(author_name: "pause_payouts_for_seller_based_on_chargeback_rate", content: "Payouts automatically paused due to chargeback rate (50%) exceeding 3% volume.")
+
+        allow(@creator).to receive(:unpaid_balance_cents).and_return(100_00)
+        expect { @creator.restore_risk_state_if_balance_recovered! }.not_to change { @creator.reload.user_risk_state }
```

Comment:
Recovered account put on probation due to another reason shouldn't be allowed to recover again